lib/index.ts

```diff
@@ -0,0 +1,572 @@
+	};
+
+	private apiKey: string;
+
+	constructor(apiKey: string) {
+		// Key.
+		this.apiKey = apiKey;
+
+		// Setup resource paths.
+		this.comment = {
+			create: _.partial(this.httpCall,
+				{ reqType: 'POST', path: 'conversations/{conversation_id}/comments' }),
+			get: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'comments/{comment_id}' }),
+			listMentions: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'comments/{comment_id}/mentions' }),
+		};
+		this.conversation = {
+			get: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'conversations/{conversation_id}' }),
+			list: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'conversations[q{q}][page={page}][limit={limit}]' }),
+			listComments: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'conversations/{conversation_id}/comments' }),
+			listFollowers: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'conversations/{conversation_id}/followers' }),
+			listInboxes: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'conversations/{conversation_id}/inboxes' }),
+			listMessages: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'conversations/{conversation_id}/messages[page={page}][limit={limit}]' }),
+			update: _.partial(this.httpCall,
+				{ reqType: 'PATCH', path: 'conversations/{conversation_id}' }),
+		};
+		this.inbox = {
+			create: _.partial(this.httpCall,
+				{ reqType: 'POST', path: 'inboxes' }),
+			get: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'inboxes/{inbox_id}' }),
+			list: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'inboxes' }, null),
+			listChannels: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'inboxes/{inbox_id}/channels' }),
+			listConversations: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'inboxes/{inbox_id}/conversations[q{q}...][page={page}][limit={limit}]' }),
+			listTeammates: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'inboxes/{inbox_id}/teammates' }),
+		};
+		this.message = {
+			get: _.partial(this.httpCall,
+				{ reqType: 'GET', path: 'messages/{message_id}' }),
+			receiveCustom: _.partial(this.httpCall,
+				{ reqType: 'POST', path: 'channels/{channel_id}/incoming_messages' }),
+			reply: _.partial(this.httpCall,
+				{ reqType: 'POST', path: 'conversations/{conversation_id}/messages' }),
+			send: _.partial(this.httpCall,
+				{ reqType: 'POST', path: 'channels/{channel_id}/messages' }),
+		};
+	}
+
+	private httpCall = (details: FrontRequest, params: any, callback?: InternalCallback): Promise<any | void> => {
+		const requestOpts = {
+			body: (params) ? params : {},
+			headers: {
+				Authorization: `Bearer ${this.apiKey}`
+			},
+			json: true,
+			method: details.reqType,
+			url: `${FrontURL}/${this.formatPath(details.path, params)}`
+		};
+
+		// Make the request.
+		return request(requestOpts).promise().catch((error: any) => {
+			// Format this into something useful.
+			throw error.error._error;
```

Comment:
@dfunckt Here's one for listing Inboxes:
```
{
    name: 'StatusCodeError',
    statusCode: 401,
    message: '401 - {"_error":{"status":401,"title":"Unauthenticated","message":"Provided token is not a JSON Web Token"}}',
    error: {
        _error: {
            status: 401,
            title: 'Unauthenticated', 
            message: 'Provided token is not a JSON Web Token'
        }
    }
  ... <further HTTPS response>
}
```
`error` on its own would be the full HTTPS response, and I don't think that's relevant. Throwing `error.error` would get you the object where you'd still have to have a level of indirection to get to the actual contents, ie. `console.log(error._error.status)` which (to me at least) would seem very annoying.

The documentation (http://dev.frontapp.com/#response-body-structure) states:
"When the response is an error, its body will contain an _error object to describe the error." and describes the body I've seen above. I've not seen a different error where they return just a message, do you have an example I can look at so I can get this right?